spec/controllers/comments_controller_spec.rb

```diff
@@ -196,6 +263,75 @@
       end
     end
 
+    context "guest comments are turned on in admin settings" do
+      let(:work) { create(:work) }
+      let(:work_with_guest_comment_off) { create(:work, comment_permissions: :disable_anon) }
+      let(:admin_setting) { AdminSetting.first || AdminSetting.create }
+
+      before do
+        admin_setting.update_attribute(:guest_comments_off, false)
+      end
+
+      it "allows guest comments" do
+        post :new, params: { work_id: work.id, comment: anon_comment_attributes }
+
+        expect(flash[:error]).to be_nil
+      end
+
+      it "does not allow guest comments when work has guest comments disabled" do
+        post :new, params: { work_id: work_with_guest_comment_off.id, comment: anon_comment_attributes }
+
+        it_redirects_to_with_error(work_path(work_with_guest_comment_off), 
+                                   "Sorry, this work doesn't allow non-Archive users to comment.")
+      end
+    end
+
+    context "guest comments are turned off in admin settings" do
+      let(:work) { create(:work) }
+      let(:admin_setting) { AdminSetting.first || AdminSetting.create }
+
+      before do
+        admin_setting.update_attribute(:guest_comments_off, true)
+      end
+
+      [:enable_all, :disable_anon].each do |permissions|
+        context "when work comment permissions are #{permissions}" do
+          before do
+            work.update_attribute(:comment_permissions, permissions)
+          end
+
+          it "redirects logged out user with an error" do
+            post :new, params: { work_id: work.id, comment: anon_comment_attributes }
+            it_redirects_to_with_error("/where_i_came_from", "Sorry, the Archive doesn't allow guests to comment right now.")
+          end
+
+          it "redirects logged in user to the comment on the commentable without an error" do
```

Comment:
I think this description might be copied from elsewhere -- the behavior here is `it "renders the :new template for logged in user"`